plugins/modules/ndo_l3out_bgp_peer.py

```diff
@@ -0,0 +1,800 @@
+        auth_password=dict(type="str", no_log=True),
+        weight=dict(type="int"),
+        site_of_origin=dict(type="str", aliases=["fabric_of_origin"]),
+        allowed_self_as_count=dict(type="int"),
+        local_asn_config=dict(type="str", choices=list(LOCAL_ASN_CONFIG)),
+        local_asn=dict(type="int"),
+        bgp_controls=dict(
+            type="dict",
+            options=dict(
+                state=dict(type="str", choices=["enabled", "disabled"]),
+                allow_self_as=dict(type="bool"),
+                override_as=dict(type="bool"),
+                disabled_peer_as_check=dict(type="bool"),
+                next_hop_self=dict(type="bool"),
+                send_community=dict(type="bool"),
+                send_extended_community=dict(type="bool"),
+                send_domain_path=dict(type="bool"),
+            ),
+        ),
+        peer_controls=dict(
+            type="dict",
+            options=dict(
+                state=dict(type="str", choices=["enabled", "disabled"]),
+                bfd=dict(type="bool"),
+                disable_peer_connected_check=dict(type="bool"),
+            ),
+        ),
+        address_families=dict(
+            type="dict",
+            options=dict(
+                state=dict(type="str", choices=["enabled", "disabled"]),
+                multicast=dict(type="bool"),
+                unicast=dict(type="bool"),
+            ),
+        ),
+        private_as_controls=dict(
+            type="dict",
+            options=dict(
+                state=dict(type="str", choices=["enabled", "disabled"]),
+                remove_all=dict(type="bool"),
+                replace_with_local_as=dict(type="bool"),
+            ),
+        ),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["template", "template_id"], True],
+            ["state", "query", ["template", "template_id"], True],
+            ["state", "absent", ["template", "template_id"], True],
+            ["state", "present", ["l3out", "l3out_uuid"], True],
+            ["state", "query", ["l3out", "l3out_uuid"], True],
+            ["state", "absent", ["l3out", "l3out_uuid"], True],
+            ["state", "present", ["ipv4_address", "ipv6_address"], True],
+            ["state", "absent", ["ipv4_address", "ipv6_address"], True],
+        ],
+        mutually_exclusive=[
+            ("template", "template_id"),
+            ("l3out", "l3out_uuid"),
+            ("import_route_map", "import_route_map_uuid"),
+            ("export_route_map", "export_route_map_uuid"),
+            ("peer_prefix", "peer_prefix_uuid"),
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template_identifier = get_object_identifier(module.params.get("template_id"), module.params.get("template"))
```

Comment:
what is the use case for template_identifier instead of just storing in two variables?